addons/SLib/SLib.gd

```diff
@@ -150,174 +169,172 @@ func full_path(path: String) -> String:
+## SLib.save_file(2, "user://Data.json", enemy_data_dict)
+## # 3: Resource (.tres, .tscn , ...), save panel style in "res://theme/custom_panel.tres":
+## SLib.save_file(3, "res://theme/custom_panel.tres", $Panel.theme_override_styles/panel)
+## [/codeblock]
+func save_file(type: file_types, location: String, value = null, config: String = "") -> void:
+	match type:
+		0:
+			var file = FileAccess.open(location,FileAccess.WRITE)
+			file.store_var(value)
+			file.close()
+		1:
+			if config.split(",", false).size() != 2:
+				send_error("Save data in config files need section & key, cann't save data in \'{file}\'".format({"file": location}), "SLib.save_file")
+				return
+			var section = config.split(",", false)[0]
+			var key = config.split(",", false)[0]
+			var config_file := ConfigFile.new()
+			config_file.set_value(section, key, value)
+			var error := config_file.save(location)
+			if error:
+				send_error("An error happened while saving data in \'{file}\' > \'{section}\' > \'{key}\': \'{error}\'".format({"file": location, "section": section, "key": key, "error": str(error)}), "SLib.save_file")
+		2:
+			var json_string := JSON.stringify(value)
+			var file_access := FileAccess.open(location, FileAccess.WRITE)
+			if not file_access:
+				send_error("An error happened while saving data in \'{file}\': \'{error}\'".format({"file": location, "error": FileAccess.get_open_error()}), "SLib.save_file")
+			file_access.store_var(json_string)
+			file_access.close()
+		3:
+			var error := ResourceSaver.save(value, location)
+			if error:
+				send_error("An error happened while saving data in \'{file}\': \'{error}\'".format({"file": location, "error": error}), "SLib.save_file")
+		_:
+			send_error("Please select a valid file type for save!", "SLib.save_file")
+
+
+## This function returns the content stored in the file, you can use it for:[br]
+## - All files created by [FileAccess] and [method save_file] with FILE_ACCESS type in [enum file_types]. (Use FILE_ACCESS for these files)[br]
+## - All config files ([code].ini[/code]). (Use CONFIG type in [enum file_types] for these files)[br]
+## - All JSON files ([code].json[/code]). (Use JSON_FILE type in [enum file_types] for these files)[br]
+## - All Resource files ([code].tres[/code], [code].tscn[/code], [code].res[/code] & [code].scn[/code]). See also [Resource] & [ResourceSaver]. (Use RESOURCE for these files)[br]
+## NOTE: [param cofig] just for COFIG [enum file_types], use [code]"%section%,%key%"[/code] pattern for this parameter. See also [method ConfigFile.get_value][br]
+## See also [method save_file].
 ## [br][br]
 ## NOTE:
-## For use project directory type [code]res://[/code] and for hidden user data directory use [code]user://[/code].
+## If the file doesn't exist, it will send an error to the console and return [param default_value].
+func load_file(type: file_types, location: String, default_value: Variant = null, config: String = "") -> Variant:
+	match type:
+		0:
```

Comment:
```suggestion
		file_types.FILE_ACCESS:
```

I think you can do this instead of putting in the actual integers. This way, it is more clear what is being matched.